src/functions.jl

```diff
@@ -48,6 +48,19 @@ for (mod, funs) in (
     end
 end
 
+if VERSION > v"1.1-"
+    function Base.eachslice(a::NamedDimsArray{L}; dims, kwargs...) where L
+        numerical_dims = dim(a, dims)
+        slices = eachslice(parent(a); dims=numerical_dims, kwargs...)
+        return Base.Generator(slices) do slice
+            # For unknown reasons (something to do with hoisting?) having this in the
+            # function passed to `Generator` actually results in less memory being allocated
+            names = remaining_dimnames_after_dropping(L, numerical_dims)
+            return NamedDimsArray(slice, names)
+        end
+    end
+end
+
```

Comment:
Do you  think this is the best file for it ?
If you want to change it then do so, or if not then don't.
(Does not block this PR from being merge)

Everything else here is more programatic.

Maybe it is better in `functions_dims` ?

Reply:
I think it fits with "standard base overloads" more than "messing with dimensions" - and fitted better in the corresponding test file -  but 🤷‍♂

 will leave for now :)